Do not run visualizations on InterruptedException

hubertp opened the pull request:
### Pull Request Description

There is no point in running visualization for the expression value that is InterruptedException. The latter is likely to bubble up the exception or create erroneous value that will be confusing to the user.

### Important Notes

Closes #11243 and partially addresses some of the symptomes of #11084.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

hubertp committed:
Do not run visualizations on InterruptException

There is no point in running visualization for the expression value that
is InterruptedException. The latter is likely to bubble up the exception
or create one that will be confusing to the user.

Closes #11243 and partially addresses some of the symptomes of #11084.

JaroslavTulach commented on engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java:
```diff
@@ -43,6 +43,14 @@ public static String findExceptionMessage(Throwable ex) {
+      return false;
+    }
+  }
+
```
Is it necessary to introduce new `isInterruptedExceptionValue` method? Isn't it easier to change the argument of existing `isInterruptedException` to `Object`?

hubertp replied:
Instance check didn't feel right, hence another method. But I don't have a preference tbh.

JaroslavTulach replied:
There is no need to perform an instance check. `isInterruptedException(Object ex, InteropLibrary iop)` accepts `Object`.

hubertp committed:
Add a test for confusing visualization failures

Previously a visualization failure would be reported:
```
Method `+` of type sleep interrupted could not be found.
```

hubertp committed:
PR review

Nit

hubertp committed:
Merge branch 'develop' into wip/hubert/11084-visualizations-on-interrupts
